Classes/Fluid/ViewHelper/ComponentRenderer.php

```diff
@@ -179,15 +203,26 @@ public function render()
                 $this->renderingContext->getTemplatePaths()->getPartialRootPaths()
             );
         }
+
         $variableContainer = $renderingContext->getVariableProvider();
+        $viewHelperVariableContainer = $renderingContext->getViewHelperVariableContainer();
+
+        // Inherit component context from parent component
+        $parentComponentContext = $viewHelperVariableContainer->get(static::class, 'componentContext');
```

Comment:
Maybe this can be optimized so that the merging and copying is only necessary in case context is actually in use?

Reply:
With the current architecture, it is probably not possible to improve this without some major refactoring.